src/cdk/testing/harness-environment.ts

```diff
@@ -165,17 +185,18 @@ export abstract class HarnessEnvironment<E> implements HarnessLoader, LocatorFac
     const skipSelectorCheck = (elementQueries.length === 0 && harnessTypes.size === 1) ||
         harnessQueries.length === 0;
 
-    const perElementMatches = await Promise.all(rawElements.map(async rawElement => {
-      const testElement = this.createTestElement(rawElement);
-      const allResultsForElement = await Promise.all(
-          // For each query, get `null` if it doesn't match, or a `TestElement` or
-          // `ComponentHarness` as appropriate if it does match. This gives us everything that
-          // matches the current raw element, but it may contain duplicate entries (e.g. multiple
-          // `TestElement` or multiple `ComponentHarness` of the same type.
-          allQueries.map(query =>
-              this._getQueryResultForElement(query, rawElement, testElement, skipSelectorCheck)));
-      return _removeDuplicateQueryResults(allResultsForElement);
-    }));
+    const perElementMatches = await this.batchCD(() =>
```

Comment:
Add a comment here the explains why you want to batch change detection here?

Reply:
This is self-explanatory now that I'm using `parallel` instead